### PR TITLE
Fix/city match manager expectations

### PR DESCRIPTION
**Pull request by flutterdevtaha**

### CODE REVIEW

**Comment by flutterdevtaha** [comment]
[Enhancement on Task 8](https://trello.com/c/klLZWukV/37-enhancement-on-task-8)

**Comment by flutterdevtaha** [comment]
@ma28alhams  I pushed everything

File: src/main/kotlin/interactor/GetCityMatchManagerExpectationsInteractor.kt [hunk 1]
```diff
@@ -1,46 +1,68 @@
...
 
 class GetCityMatchManagerExpectationsInteractor(
     private val dataSource: CostOfLivingDataSource,
 ) {
 
-    private fun getOnlyNorthAmericaCountries(city: CityEntity): Boolean =
-        city.country in NorthAmericaCountries.list
+    fun execute(): CityEntity? {
+        return dataSource.getAllCitiesData()
+            .let { it.ifEmpty { throw NoReturnedDataException(Constants.EMPTY_LIST_EXCEPTION) } }
```
Comment (devsadeq) [inline src/main/kotlin/interactor/GetCityMatchManagerExpectationsInteractor.kt:13]: so should we use also or let in this case ?

**Review by devsadeq — APPROVED** [review]
well done.

**Review by HaidyAbuGom3a — APPROVED** [review]

File: src/main/kotlin/interactor/GetCityMatchManagerExpectationsInteractor.kt (hunk 1 above)
Comment (devsadeq) [reply src/main/kotlin/interactor/GetCityMatchManagerExpectationsInteractor.kt:13]: btw u can use ifEmpty directly 
ex:
.ifEmpty { throw NoReturnedDataException(EMPTY_DATA_SOURCE_EXCEPTION_MSG) }

**Review by devsadeq — APPROVED** [review]

File: src/main/kotlin/interactor/GetCityMatchManagerExpectationsInteractor.kt (hunk 1 above)
Comment (flutterdevtaha) [reply src/main/kotlin/interactor/GetCityMatchManagerExpectationsInteractor.kt:13]: nice note.

**Pull request merged**